tools/walletextension/userconn/user_conn.go

```diff
@@ -61,15 +61,19 @@ func NewUserConnWS(resp http.ResponseWriter, req *http.Request) (UserConn, error
 func (h *userConnHTTP) ReadRequest() ([]byte, error) {
 	body, err := io.ReadAll(h.req.Body)
 	if err != nil {
-		return nil, fmt.Errorf("could not read request body: %w", err)
+		wrappedErr := fmt.Errorf("could not read request body: %w", err)
+		h.HandleError(wrappedErr.Error())
```

Comment:
Here and below, I've changed the API so that the UserConn is in charge of handling the error (i.e. logging, printing, and writing it back to the websocket), rather than expecting the call of `ReadRequest`/`WriteResponse` to call `HandleError` themselves.